app/background.ts

```diff
@@ -59,14 +63,19 @@ class BackgroundWorker {
 
   callback(details:any) {
     const tabRecords = this.data[this.currentTab];
-    if (tabRecords.enabled && this.currentTab === details.tabId) {
-      tabRecords.requests.push(details);
-      this.data[this.currentTab] = tabRecords;
-    }
+    tabRecords.requests.slice(0)  //possible fix for displaying proper badge text 
```

Comment:
why we need to do this? `.slice` won't update the original array.

Reply:
Yea, realised that and used ```tabRecords.requests = []``` . Will commit that